Tiber upgrade (#259)
autonity/docs.autonity.org: oracle/run-oracle/index.md

The runtime configuration of oracle server can be specified using a configuration file, command line flags, or system environment variables. Using the configuration file `oracle-server.config` is the preferred default path.

### Set up oracle server config file

The oracle server config file `oracle-server.config` can be found in the `/autonity-oracle/config` sub-directory.  Edit the file to set the config values where:

> | Option | Description |

Reviewer: Indeed, a table makes the options much easier to review

Author: Yup.

| `key.file` | Path to the oracle server key file. e.g. `../aut/keystore/oracle.key`|
| `key.password` | The password to the oracle server key file. |
| `log.level` | Sets logging verbosity. Available logging levels: `0`: No logging, `1`: Trace, `2`: Debug, `3`: Info, `4`: Warn, `5`: Error |
| `ws` | The WS-RPC server listening interface and port of the connected Autonity Go Client node (see [install Autonity, networks](/node-operators/install-aut/#network). E.g.: `"ws://127.0.0.1:8546"` |
| `plugin.dir` | The path to the directory containing the built data plugins. For example `"./plugins"` |
| `plugin.conf` | The path to the plugins configuration file. For example `"./plugins-conf.yml"` |
| `confidence.strategy` | The confidence rule. Available strategies are: `0`: linear, `1`: fixed. |
| `profile.dir` | The profiling report directory, where the profiling report (i.e. runtime state) will be saved to. For example `"."` |

::: {.callout-tip title="Confidence strategy - linear vs. fixed" collapse="false"}

> Oracle prices are submitted with a _confidence score_ in the range $(0, 100)$ expressing the oracle's level of trust in the provided price (For detail on confidence score, see the concept description [Oracle accountability fault detection (OAFD)](/concepts/oafd/#confidence-score).

Reviewer 2: ```suggestion
Oracle prices are submitted with a _confidence score_ in the range $(0, 100]$ expressing the oracle's level of trust in the provided price (For detail on confidence score, see the concept description [Oracle accountability fault detection (OAFD)](/concepts/oafd/#confidence-score).
```
100 confidence is allowed

- Crypto plugins: for connecting to public CEX and DEX data sources for USD stablecoin and ATN, NTN prices. ATN NTN price data is used for the ASM [Stabilisation CDP](/concepts/asm/#stabilization) mechanism. See the `crypto_` prefixed adaptors in [`/plugins`](https://github.com/autonity/autonity-oracle/tree/master/plugins). Four crypto plugins are currently provided.
- Simulator plugin: for simulated protocol asset (ATN, NTN, NTN-ATN) data. Used for testnet or local development purposes to provide ATN, NTN price data. See the `simulator_plugin` adaptor in [`/plugins`](https://github.com/autonity/autonity-oracle/tree/master/plugins).

Plugins are configured by default or by explicit configuration. The `crypto_` plugins have a default configuration set in the plugin source golang code as a `defaultConfig`. The `forex_` and `simulator_` plugin configuration is specified explicitly in the `plugins-conf.yml` file by addding a config entry for each plugin configured.

The full set of plugin configuration fields are:

> | Name | Datatype | Mandatory? | Description |

Reviewer: In this table, consider using checkbox ✓ Unicode character for "required" and omitting the checkbox to mean "optional".

Author: Done. There's an interesting GitHub gist [Checkbox Tables in Markdown](https://gist.github.com/TheMatt2/1360af52468714370c753cfa2fe50783).

I've used the HTML entity approach - &#x2714;

Commit 62a81af5ffee2cc164232ff9f604eac6de3b035e

It is better practice to retrieve ATN, NTN price data from on-chain using your own hosted validator node rather than testnet infrastructure.
:::

Customise the `crypto_uniswap`plugin:

1. Edit the [`defaultConfig`](https://github.com/autonity/autonity-oracle/blob/v0.2.3/plugins/crypto_uniswap/crypto_uniswap.go#L26) in the source golang file must be edited to change:

    - [`defaultConfig.Endpoint`](https://github.com/autonity/autonity-oracle/blob/v0.2.3/plugins/crypto_uniswap/crypto_uniswap.go#L28): change the `endpoint` to replace the `rpc1-internal` endpoint with the endpoint address of their own connected validator node.
    - [`defaultConfig.Scheme`](https://github.com/autonity/autonity-oracle/blob/v0.2.3/plugins/crypto_uniswap/crypto_uniswap.go#L29): update to the scheme used for the endpoint (http/s or ws/s).

> 2. Build the plugin. If building oracle server from binary, then customise the plugin before running `make autoracle`. Alternatively, build the plugin individually a describ ed in [/plugins/README, Build it](https://github.com/autonity/autonity-oracle/blob/v0.2.3/plugins/README.md#build-it)]. In the `autonity-oracle` directory run:

Reviewer 2: Not sure if this has already been addressed, but this should definitely be a config parameter. Asking node operators to modify the source code and recompile is not ideal.

Author: FYSA @Jason-Zhangxin-Chen 

@lorenzo-dev1 - it's current state so the docs should reflect that state.

Can you create this as an issue in the autonity-oracle repo and resolve here?

Then, the docs can be updated once the issue is resolved.

Author: But there is an extra space and typo that needs correcting:

>...build the plugin individually a describ ed ...

That should be 'as described'

Author: Which typo then got fixed in the following comment.

Reviewer 2: issue created https://github.com/autonity/autonity-oracle/issues/58, solving this thread then.

Reviewer 3: > FYSA @Jason-Zhangxin-Chen
> 
> @lorenzo-dev1 - it's current state so the docs should reflect that state.
> 
> Can you create this as an issue in the autonity-oracle repo and resolve here?
> 
> Then, the docs can be updated once the issue is resolved.

Those are default configs, people can config them from the config file by adding: 
```yaml
# Un-comment below lines to config the RPC endpoint of a Piccadilly Network Full Node for your AMM plugin which sources ATN & NTN market data from an on-chain AMM.
#  - name: crypto_uniswap
#    scheme: "wss"                                          # Only websocket please, available values are: "ws" or "wss", default value is "wss" for uniswap plugins.
#    endpoint: "rpc-internal-1.piccadilly.autonity.org/ws"  # The default URL might not be stable for public usage, we recommend you to change it with your validator node's RPC endpoint.
```

Author: Right, this is the case in v0.2.3. Edited to simplify and reword in commit 5d40885fdf40c390de89614d392aa4c020100d66

---

This section removed in its entirety:

```
Default configuration is specified in the crypto plugins' source golang files in a `defaultConfig` structure. To change the default configuration that structure needs to be edited before building the plugin.

::: {.callout-important title="Only the DEX `crypto_uniswap` plugin requires customisation of the default configuration" collapse="true"}
The validator operator must re-configure the `crypto_uniswap` connector to provide data to their own validator node.

This is because the `crypto_uniswap`plugin [`defaultConfig`](https://github.com/autonity/autonity-oracle/blob/v0.2.3/plugins/crypto_uniswap/crypto_uniswap.go#L26) is set to a testing configuration using an `rpc1-internal` endpoint on the Piccadilly Testnet. If the configured endpoint is not changed, then the plugin will be configured to take data from the rpc1-internal testnet node and not your own validator node state.

It is better practice to retrieve ATN, NTN price data from on-chain using your own hosted validator node rather than testnet infrastructure.
:::

Customise the `crypto_uniswap`plugin:

1. Edit the [`defaultConfig`](https://github.com/autonity/autonity-oracle/blob/v0.2.3/plugins/crypto_uniswap/crypto_uniswap.go#L26) in the source golang file must be edited to change:

    - [`defaultConfig.Endpoint`](https://github.com/autonity/autonity-oracle/blob/v0.2.3/plugins/crypto_uniswap/crypto_uniswap.go#L28): change the `endpoint` to replace the `rpc1-internal` endpoint with the endpoint address of their own connected validator node.
    - [`defaultConfig.Scheme`](https://github.com/autonity/autonity-oracle/blob/v0.2.3/plugins/crypto_uniswap/crypto_uniswap.go#L29): update to the scheme used for the endpoint (http/s or ws/s).

2. Build the plugin. If building oracle server from source, then customise the plugin before running `make autoracle`. Alternatively, build the plugin individually as described in [/plugins/README, Build it](https://github.com/autonity/autonity-oracle/blob/v0.2.3/plugins/README.md#build-it). In the `autonity-oracle` directory run:
   
     ```yaml
     go build -o ./build/bin/plugins/crypto_uniswap ./plugins/crypto_uniswap/crypto_uniswap.go
     ```

The other crypto CEX plugins require no customisation. No additional action is required. These can be run run out the box as-is:

- `crypto_kraken`
- `crypto_coingecko`
- `crypto_coinbase`

```

> 2. Build the plugin. If building oracle server from binary, then customise the plugin before running `make autoracle`. Alternatively, build the plugin individually a describ ed in [/plugins/README, Build it](https://github.com/autonity/autonity-oracle/blob/v0.2.3/plugins/README.md#build-it)]. In the `autonity-oracle` directory run:

Reviewer 2: ```suggestion
Customise the `crypto_uniswap`plugin:

1. Edit the [`defaultConfig`](https://github.com/autonity/autonity-oracle/blob/v0.2.3/plugins/crypto_uniswap/crypto_uniswap.go#L26) in the source golang file must be edited to change:

    - [`defaultConfig.Endpoint`](https://github.com/autonity/autonity-oracle/blob/v0.2.3/plugins/crypto_uniswap/crypto_uniswap.go#L28): change the `endpoint` to replace the `rpc1-internal` endpoint with the endpoint address of their own connected validator node.
    - [`defaultConfig.Scheme`](https://github.com/autonity/autonity-oracle/blob/v0.2.3/plugins/crypto_uniswap/crypto_uniswap.go#L29): update to the scheme used for the endpoint (http/s or ws/s).

2. Build the plugin. If building oracle server from source, then customise the plugin before running `make autoracle`. Alternatively, build the plugin individually as described in [/plugins/README, Build it](https://github.com/autonity/autonity-oracle/blob/v0.2.3/plugins/README.md#build-it). In the `autonity-oracle` directory run:
```

   
     ```yaml
     go build -o ./build/bin/plugins/crypto_uniswap ./plugins/crypto_uniswap/crypto_uniswap.go
     ```

The other crypto CEX plugins require no customisation. No additional action is required. These can be run run out the box as-is:

- `crypto_kraken`
- `crypto_coingecko`
- `crypto_coinbase`

### ATN and NTN data simulator plugin

> If you are setting up or connecting to a local Autonity testnet an ATN and NTN Simulator plugin is available.

Reviewer 2: I feel like this sentence could be removed, i am not really sure what it tries to communicate

Author: To make it clear to a user when you use the simulator - in a testnet scenario, either on a public testnet or in your own local testnet.

Removed. b6441d159c2d882af8c4174b1d91495014be3598

> If a simulator has been deployed and is available to provide simulated data for testnet use, this can be connected to by adding a config entryfor the simulated source to your `plugins-conf.yml` file.

Reviewer 2: ```suggestion
If a simulator has been deployed and is available to provide simulated data for testnet use, this can be connected to by adding a config entry for the simulated source to your `plugins-conf.yml` file.
```
